core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -1003,8 +1003,12 @@ private static void tryWriteTempFile(Path path) throws IOException {
         if (Files.exists(path)) {
             Path resolve = path.resolve(".es_temp_file");
             try {
-                Files.createFile(resolve);
```

Comment:
I wonder if it'd be more right to delete the file if it exists, then create it, then delete it again. That way even if we start with a leftover file we'll create *and* delete the file. I can just imagine getting some esoteric failure in the future because we could write to a directory but not create files in it in some system with a more complex permissions scheme. I know that is totally paranoid though.

Could you also add a comment to the code about this handling the case when the file is left over from a crash?